src/sbt-test/sbt-guardrail/java-codegen-app/build.sbt

```diff
@@ -0,0 +1,28 @@
+
+enablePlugins(GuardrailPlugin)
+
+name := "sbt-guardrail-java-test-app"
+
+version := "1.0." + System.currentTimeMillis
+
+scalaVersion := "2.12.10"
+
+scalacOptions += "-Xexperimental"
+
+guardrailTasks in Compile := List(
+  JavaClient(file("petstore.json"), pkg="com.example.clients.petstore")
+)
+
+// workaround for SBT issue.  see: https://github.com/sbt/sbt/issues/1664
+unmanagedSourceDirectories in Compile += (sourceManaged in Compile).value
```

Comment:
Should this be `managedSourceDirectories in Compile`?

Reply:
I don't have context on why this construct was chosen, but as managed seems to be the way to go, considering the linked issue, probably yes.

As this is just in the scripted tests, I wonder what the actual impact is -- maybe accumulating cruft when folks run the scripted tests locally?

Reply:
The reason I questioned it is because I saw it from the linked issue and it confused me as I was trying to figure out what to do in my own project. I wanted to make sure there wasn't a reason for doing it this way

Reply:
Yeah, I saw your contribution to the thread -- I must admit I don't really understand the distinction myself. I'm really only drawn into sbt-guardrail when something actively breaks, as there's still much to do in the primary repo.

Looking at https://github.com/sbt/sbt/issues/4654, it seems as though `managedSourceDirectories` really is the correct thing to do, though due to the simplicity of these tests (not integrating with any other plugins) I don't think there's a _huge_ material difference to these tests.

Reply:
Yeah, I hadn't realized this was test code, but as long as the tests are passing it doesn't matter. For reference, "managed" means it's generated code whereas "unmanaged" is just regular user-written code

Reply:
In that case, it should be changed for correctness, thanks for bringing it up!